File: src/main/java/io/beaniejoy/dongnecafe/cafe/repository/CafeRepository.java
package io.beaniejoy.dongnecafe.cafe.repository;

import io.beaniejoy.dongnecafe.cafe.domain.Cafe;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.UUID;

public interface CafeRepository extends JpaRepository<Cafe, UUID> {

    Page<Cafe> findAll(Pageable pageable);
}

File: db/src/main/kotlin/io/beaniejoy/dongnecafe/db/cafe/repository/CafeRepository.kt
package io.beaniejoy.dongnecafe.db.cafe.repository

import io.beaniejoy.dongnecafe.db.cafe.repository.custom.CafeQueryRepository
import io.beaniejoy.dongnecafe.domain.cafe.entity.Cafe
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface CafeRepository : JpaRepository<Cafe, Long>, CafeQueryRepository {
    fun findByName(name: String): Cafe?

    fun findByNameContainingIgnoreCase(name: String?, pageable: Pageable): Page<Cafe>
}
